test/test_helper.rb

```diff
@@ -110,6 +113,58 @@ def fake_machine(options = {})
   machine
 end
 
+class FakeHTTPProxyServer < WEBrick::HTTPProxyServer
```

Comment:
I suggest we put this into its own class. test_helper has already so much stuff and we might need this later for feature tests as well.